module/registry.go

```diff
@@ -0,0 +1,59 @@
+package module
+
+import (
+	"sync"
+)
+
+var modules = make(map[string]NewModule)
+var modulesLck sync.RWMutex
```

Comment:
Typo: "lock"?

Reply:
I do have a bad habit of dropping letters from commonly used words) Will change to "lock".